app/controllers/proposal/collection_draft_proposals_controller.rb

```diff
@@ -2,7 +2,45 @@ module Proposal
   class CollectionDraftProposalsController < CollectionDraftsController
     skip_before_action :provider_set?
 
+    # TODO: Limit this to only the things a user is supposed to do, also need a new one for approver
+    # TODO: Also need one for functions which both can perform (e.g. show)
     before_action :ensure_non_nasa_draft_user
+    before_action(only: %I[submit rescind]) { set_resource }
+
+    def edit
+      unless get_resource&.in_work?
+        flash[:error] = 'Only proposals in an "In Work" status can be edited.'
+        redirect_to collection_draft_proposal_path(get_resource) and return
```

Comment:
I think this might be a _little better_ and we can omit the `and return`:
```
if !get_resource&.in_work?
  super
else
  #{your_changes (without `and return`)}
end
```